ament_cmake_pytest/cmake/ament_add_pytest_test.cmake

```diff
@@ -75,8 +75,9 @@ function(ament_add_pytest_test testname path)
   ament_has_pytest(has_pytest QUIET PYTHON_EXECUTABLE "${ARG_PYTHON_EXECUTABLE}")
   if(NOT has_pytest)
     message(WARNING
-      "The Python module 'pytest' was not found, pytests can not be run (e.g. "
-      "on Ubuntu/Debian install the package 'python3-pytest')")
+      "The Python module 'pytest' was not found, pytests cannot be run. "
+      "On Ubuntu/Debian, install the 'python3-pytest' package. "
+      "On other platforms, install 'pytest' using pip.")
```

Comment:
another thing: should `ament` really link to ROS 2 like that? As opposed to the other way around, ROS 2 linking to `ament`.

Reply:
> I kind of prefer the current warning messages because they tell you exactly what you need to do (install ___ package) instead of pointing you to the generic installation docs just to install a Python package or two.

Yeah, that's true.

> another thing: should `ament` really link to ROS 2 like that? As opposed to the other way around, ROS 2 linking to `ament`.

In theory, no.  In practice, it seems unlikely that anyone outside the ROS 2 ecosystem is using ament (at least, I don't know of any).  It would be nice to be proven wrong, though :).

Overall, I agree with you about keeping the current message.  The one thing I would say is let's change the message to say "On Linux, install the 'python3-pytest' package", since that will also cover RHEL/Fedora.

Reply:
> In theory, no. In practice, it seems unlikely that anyone outside the ROS 2 ecosystem is using ament (at least, I don't know of any). It would be nice to be proven wrong, though :).

maybe one day! :smile:

> Overall, I agree with you about keeping the current message. The one thing I would say is let's change the message to say "On Linux, install the 'python3-pytest' package", since that will also cover RHEL/Fedora.

done: 0752f311591066c35dc8136a044a0e1d2781ada1